core/story-api/src/main/java/org/lgna/story/SJointedModel.java

```diff
@@ -59,25 +61,39 @@ public abstract class SJointedModel extends SModel {
   @MethodTemplate(visibility = Visibility.COMPLETELY_HIDDEN)
   public abstract JointedModelImp getImplementation();
 
+  // Cache skeleton for easy access
+  private final InitializingIfAbsentMap<JointId, SJoint> jointsById =  Maps.newInitializingIfAbsentHashMap();
+  private final InitializingIfAbsentMap<String, SJoint> jointsByName = Maps.newInitializingIfAbsentHashMap();
+  private final InitializingIfAbsentMap<JointId[], SJoint[]> jointArrays = Maps.newInitializingIfAbsentHashMap();
+
   //todo: make protected
   @MethodTemplate(visibility = Visibility.COMPLETELY_HIDDEN)
   public SJoint getJoint(JointId jointId) {
-    return SJoint.getJoint(this, jointId);
+    return jointsById.getInitializingIfAbsent(jointId, id -> getImplementation().getJointImplementation(id).getAbstraction());
```

Comment:
this is feeling extremely nitpicky, but I guess it isn't surprising that a InitializingIfAbsentMap will initialize something if it is absent and maybe  it could just be jointsById.get(jointId, X) ?

Reply:
That seems better to me, butI will make it a distinct PR.